src/compute/src/sink/persist_sink.rs

```diff
@@ -1035,9 +1083,19 @@ where
             for done_batch_metadata in done_batches.into_iter() {
                 in_flight_descriptions.remove(&done_batch_metadata);
 
-                let mut batches = in_flight_batches
+                let batch_set = in_flight_batches
                     .remove(&done_batch_metadata)
-                    .unwrap_or_else(Vec::new);
+                    .unwrap_or_default();
+
+                let mut batches = batch_set.finished;
+                if !batch_set.incomplete.is_empty() {
+                    let (lower, upper) = done_batch_metadata.clone();
+                    let batch = write
+                        .batch(batch_set.incomplete, lower, upper)
```

Comment:
we could also put it in a BatchBuilder as we go, which shouldn't really have any overhead (though I'd sanity check this quickly before committing to it). if not, at least leave this as a TODO, please

Reply:
Done; seems AOK. (Though it does prevent us from consolidating here...)